src/TLuaInterpreter.cpp

```diff
@@ -16739,3 +16739,246 @@ int TLuaInterpreter::getDictionaryWordList(lua_State* L)
 
     return 1;
 }
+
+// Internal function - helper for updateColorTable().
+void TLuaInterpreter::insertColorTableEntry(lua_State* L, const QColor& color, const QString& name)
+{
+    // Equivalent (when called from updateColorTable()) to Lua (where the
+    // '<8-bit unsigned int, i.e. 0 to 255>'s are provided from the QColor):
+    // color_table["name"] = { <color.red()>, <color.green()>, <color.blue()> }
+
+    // Creates a new empty table on the stack with space preallocated for 3
+    // array elements and 0 non-array elements:
+    lua_createtable(L, 3, 0);
+
+    lua_pushnumber(L, color.red());
+    lua_rawseti(L, -2, 1);
+
+    lua_pushnumber(L, color.green());
+    lua_rawseti(L, -2, 2);
+
+    lua_pushnumber(L, color.blue());
+    lua_rawseti(L, -2, 3);
+
+    lua_getfield(L, LUA_GLOBALSINDEX, "color_table");
+    lua_insert(L, -2);
```

Comment:
If possible, I'd like to get rid of these `lua_insert`s that reorder the stack. You could first get the `color_table`, then push the name to the stack and the build the colour component table.

Reply:
To be honest this code is very cargo-cult - I based it on the code produced by converting a simple lua script to C equivalent (via [lua2c](https://github.com/davidm/lua2c/)) and I really had a hard job getting my head around (and in the end did not :confounded:) the stack manipulations involved - it worked - so I was just glad for that without worrying too much about the *effectiveness* of it... :man_shrugging:

Reply:
Yes, but it's even harder to wrap your head around the lua stack manipulation this way if you suddenly start reordering the stack. So pretty please?